protocol/swap/swap_manager.go

```diff
@@ -140,40 +147,86 @@ func (m *manager) GetPastSwap(id types.Hash) (*Info, error) {
 	return s, nil
 }
 
-// GetOngoingSwap returns the ongoing swap's *Info, if there is one.
-func (m *manager) GetOngoingSwap(id types.Hash) (Info, error) {
+// GetOngoingSwap returns the ongoing swap's *Info, if there is one. The
+// returned Info structure of an active swap can be modified as the swap's state
+// changes and should only be read or written by a single go process.
+func (m *manager) GetOngoingSwap(offerID types.Hash) (*Info, error) {
 	m.RLock()
 	defer m.RUnlock()
-	s, has := m.ongoing[id]
+
+	s, has := m.ongoing[offerID]
 	if !has {
-		return Info{}, errNoSwapWithID
+		return nil, errNoSwapWithOfferID
 	}
 
-	return *s, nil
+	return s, nil
 }
 
-// GetOngoingSwaps returns all ongoing swaps.
-func (m *manager) GetOngoingSwaps() ([]*Info, error) {
+// GetOngoingSwapSnapshot returns a copy of the ongoing swap's Info, if the
+// offerID has an ongoing swap.
+func (m *manager) GetOngoingSwapSnapshot(offerID types.Hash) (*Info, error) {
 	m.RLock()
 	defer m.RUnlock()
-	swaps := make([]*Info, len(m.ongoing))
-	i := 0
+
+	s, has := m.ongoing[offerID]
+	if !has {
+		return nil, errNoSwapWithOfferID
+	}
+
+	// We have the read lock above when making this copy, but the swapState
+	// instances don't use that lock when they modify the same Info structure
+	// from a different go process. At some point we should update the swapState
+	// instances to make all changes to their ongoing Info structure via this
+	// manager.
+	sc, err := s.DeepCopy()
+	if err != nil {
+		return nil, err
+	}
+
+	return sc, nil
+}
+
+// GetOngoingSwapOfferIDs returns a list of the offer IDs of all ongoing
+// swaps.
+func (m *manager) GetOngoingSwapOfferIDs() ([]*types.Hash, error) {
+	m.RLock()
+	defer m.RUnlock()
+
+	offerIDs := make([]*types.Hash, 0, len(m.ongoing))
 	for _, s := range m.ongoing {
-		sCopy := new(Info)
-		*sCopy = *s
-		swaps[i] = sCopy
-		i++
+		offerIDs = append(offerIDs, &s.OfferID)
 	}
+
+	return offerIDs, nil
+}
+
+// GetOngoingSwapsSnapshot returns a copy of all ongoing swaps. If you need to
+// modify the result or read from its status channel, call `GetOngoingSwap` on
```

Comment:
`Info` no longer has a status channel?

Reply:
Correct. There is some information in the PR description as to why.

Reply:
yeah I meant you need to update the comment here

Reply:
Now I understand. I removed the part about the status channel from the comment. Thanks.